src/backends/pgfplotsx.jl

```diff
@@ -206,41 +206,52 @@ function (pgfx_plot::PGFPlotsXPlot)(plt::Plot{PGFPlotsXBackend})
             for series in series_list(sp)
-                    if hascolorbar(series)
-                        cg = get_colorgradient(series)
-                        cm = pgfx_colormap(get_colorgradient(series))
-                        PGFPlotsX.push_preamble!(
-                            pgfx_plot.the_plot,
-                            """\\pgfplotsset{
-                            colormap={plots$(sp.attr[:subplot_index])}{$cm},
-                            }""",
-                        )
+                if hascolorbar(series)
+                    cg = get_colorgradient(series)
+                    cm = pgfx_colormap(get_colorgradient(series))
+                    PGFPlotsX.push_preamble!(
+                        pgfx_plot.the_plot,
+                        """\\pgfplotsset{
+                        colormap={plots$(sp.attr[:subplot_index])}{$cm},
+                        }""",
+                    )
+                    push!(axis_opt, "colormap name" => "plots$(sp.attr[:subplot_index])")
+                    if cg isa PlotUtils.CategoricalColorGradient
                         push!(
                             axis_opt,
-                            "colorbar" => nothing,
-                            "colormap name" => "plots$(sp.attr[:subplot_index])",
+                            "colormap access" => "piecewise const",
+                            "colorbar sampled" => nothing,
                         )
-                        if cg isa PlotUtils.CategoricalColorGradient
-                            push!(
-                                axis_opt,
-                                "colormap access" => "piecewise const",
-                                "colorbar sampled" => nothing,
-                            )
-                        end
-                        # goto is needed to break out of col and series for
-                        @goto colorbar_end
                     end
+                    # goto is needed to break out of col and series for
+                    @goto colorbar_end
+                end
             end
             @label colorbar_end
 
-            push!(
-                axis_opt,
-                "colorbar style" => PGFPlotsX.Options(
+            if hascolorbar(sp)
+                xcstr = plot_color(sp[:xaxis][:tickfontcolor])
+                ycstr = plot_color(sp[:yaxis][:tickfontcolor])
+                colorbar_style = PGFPlotsX.Options(
                     "title" => sp[:colorbar_title],
-                ),
-                "point meta max" => get_clims(sp)[2],
-                "point meta min" => get_clims(sp)[1],
-            )
+                    "xticklabel style" => pgfx_get_ticklabel_style(sp, sp[:xaxis]),
+                    "yticklabel style" => pgfx_get_ticklabel_style(sp, sp[:yaxis]),
+                )
+                if sp[:colorbar] === :top
+                    push!(colorbar_style,
+                        "at" => string((0.5, 1.05)),
```

Comment:
I think I used `1.02` everywhere else in the code, so maybe use it here also or does it look ugly?

Reply:
From my quick comparisons 1.05 looks rather like the margin for `colorbar = :right`

Reply:
also fine

Reply:
for reference in the manual is stated that `colorbar right` defines `/pgfplots/colorbar shift/.style={xshift=0.3cm},`

Reply:
anyway, that can wait, until there are problems with the way it is in this PR

Reply:
Yes I also do not yet consider subplot titles for `colorbar = :top`. But I agree, this can be fixed in a separate PR.